tc_aws/loaders/s3_loader.py

```diff
@@ -32,6 +34,11 @@ def load(context, url, callback):
         def handle_data(file_key):
             if not file_key or 'Error' in file_key or 'Body' not in file_key:
                 logger.warn("ERROR retrieving image from S3 {0}: {1}".format(key, str(file_key)))
+                # We will now handle the failure and return a 404.
+                result = LoaderResult()
```

Comment:
Hmmm we could callback with an error, but maybe not always return the specific S3 error code to S3. A 403 on AWS side may not have the same meaning than a 403 on the Thumbor side... I think you could test for 404, in that case return an ERROR_NOT_FOUND, otherwise return an ERROR_UPSTREAM. What do you think about that?